cmd/wallet.go

```diff
@@ -100,33 +139,123 @@ It prints the accounts from the wallet file.`,
+		}
+
+		t := table.NewWriter()
+		t.SetOutputMirror(os.Stdout)
+		t.SetTitle("Wallet Contents")
+		caption := ""
+		if printPrivate {
+			caption = fmt.Sprintf("Mnemonic: %s", w.Mnemonic())
+		}
+		if !printFull {
+			if printPrivate {
+				caption += "\n"
+			}
+			caption += "To print full keys, use the --full flag."
+		}
+		t.SetCaption(caption)
+		maxWidth := 20
+		if printFull {
+			// full key is 64 bytes which is 128 chars in hex, need to print at least this much
+			maxWidth = 150
+		}
+		// TODO: add spacemesh address format (bech32)
+		if printPrivate {
+			t.AppendHeader(table.Row{
+				"pubkey",
+				"privkey",
+				"path",
+				"name",
+				"created",
+			})
+			t.SetColumnConfigs([]table.ColumnConfig{
+				{Number: 1, WidthMax: maxWidth, WidthMaxEnforcer: widthEnforcer},
+				{Number: 2, WidthMax: maxWidth, WidthMaxEnforcer: widthEnforcer},
+			})
+		} else {
+			t.AppendHeader(table.Row{
+				"pubkey",
+				"path",
+				"name",
+				"created",
+			})
+			t.SetColumnConfigs([]table.ColumnConfig{
+				{Number: 1, WidthMax: maxWidth, WidthMaxEnforcer: widthEnforcer},
+			})
+		}
+
+		// print the master keypair
+		master := w.Secrets.MasterKeypair
+		encoder := hex.EncodeToString
+		if printBase58 {
+			encoder = base58.Encode
+		}
+		if master != nil {
+			if printPrivate {
+				t.AppendRow(table.Row{
+					encoder(master.Public),
+					encoder(master.Private),
+					master.Path.String(),
+					master.DisplayName,
+					master.Created,
+				})
+			} else {
+				t.AppendRow(table.Row{
+					encoder(master.Public),
+					master.Path.String(),
+					master.DisplayName,
+					master.Created,
+				})
+			}
+		}
+
 		for _, a := range w.Secrets.Accounts {
-			err := json.NewEncoder(os.Stdout).Encode(a)
-			cobra.CheckErr(err)
-			fmt.Println()
+			if printPrivate {
+				t.AppendRow(table.Row{
+					encoder(a.Public),
+					encoder(a.Private),
+					a.Path.String(),
+					a.DisplayName,
+					a.Created,
+				})
+			} else {
+				t.AppendRow(table.Row{
+					encoder(a.Public),
+					a.Path.String(),
+					a.DisplayName,
+					a.Created,
+				})
+			}
 		}
+		t.Render()
 	},
 }
 
 func init() {
 	rootCmd.AddCommand(walletCmd)
 	walletCmd.AddCommand(createCmd)
 	walletCmd.AddCommand(readCmd)
-
-	// Here you will define your flags and configuration settings.
-
-	// Cobra supports Persistent Flags which will work for this command
-	// and all subcommands, e.g.:
-	// walletCmd.PersistentFlags().String("foo", "", "A help for foo")
-
-	// Cobra supports local flags which will only run when this command
-	// is called directly, e.g.:
-	// walletCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
+	readCmd.Flags().BoolVarP(&printPrivate, "private", "p", false, "Print private keys")
+	readCmd.Flags().BoolVarP(&printFull, "full", "f", false, "Print full keys (no abbreviation)")
+	readCmd.Flags().BoolVar(&printBase58, "base58", false, "Print keys in base58 (rather than hex)")
 }
```

Comment:
I think this removes the need for the extra parsing later:

```suggestion
	readCmd.PersistentFlags().BoolVarP(&debug, "debug", "d", false, "enable debug mode")
}
```

The `debug` flag defined in `root.go` doesn't seem to be used anywhere and could be used here instead, this would remove the need for the extra parsing above.